streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractStateManager.java

```diff
@@ -45,8 +46,7 @@
     OffsetCheckpoint checkpoint;
 
     final Map<TopicPartition, Long> checkpointableOffsets = new HashMap<>();
-    final Map<String, StateStore> stores = new LinkedHashMap<>();
-    final Map<String, StateStore> globalStores = new LinkedHashMap<>();
+    final FixedOrderMap<String, Optional<StateStore>> globalStores = new FixedOrderMap<>();
```

Comment:
Minor comment after looking over the whole PR, do we need a `FixedOrderMap<String, Optional<StateStore>>` vs just `FixedOrderMap<String, StateStore>`?

IIRC we can't pass in a `null` state store and from what I can see it's the `GlobalStateManager` and `ProcessorStateManager` that put an `Optional.empty()` in the map vs. clearing the map in the respective `close` methods.

Not a big deal, IMHO it added a little complexity to the code and I'm not sure what we gain with it.

Reply:
I'm sorry, but I'm not quite getting the feedback. The GlobalStateManager passes this collection into `AbstractStateManager#reinitializeStateStoresForPartitions`, which may de-register some of the stores, just like what happens with the ProcessorStateManager. Or did you mean something else?